tools/autograd/templates/Functions.cpp

```diff
@@ -146,8 +146,19 @@ Tensor pow_backward_self(Tensor grad, const Tensor & self, const Tensor & expone
   return at::where(exponent == 0.0, at::zeros({}, grad.options()), grad * exponent * self.pow(exponent - 1));
 }
 
-Tensor pow_backward_exponent(Tensor grad, const Tensor & self, Tensor result) {
-  return at::where(self == 0, at::zeros({}, grad.options()), grad * result * self.log());
+// Caveats:
+// We define d(a^b)/db at a = 0 and b < 0 to be -inf. This is due to
+// * d(a^b)/db -> -inf for a fixed b as a -> +0
+// * convenience of implementation: no need to have a second at::where statement
+// Currently, tensorflow defines d(a^b)/db = nan for a = 0 and b < 0.
+//
+// We define d(a^b)/db = 0 for a = 0 and b = 0 by continuity as
+// d(a^b)/db = 0 for a = 0 and b > 0.
```

Comment:
I think you want `a -> +0` here, not `a = 0`.

Reply:
I thought we wanted continuity along a = 0 while b -> +0?